log4j-core/src/main/java/org/apache/logging/log4j/core/message/ExtendedThreadInformation.java

```diff
@@ -174,4 +174,14 @@ private void formatState(final StringBuilder sb, final ThreadInfo info) {
                 break;
         }
     }
+
+    private static StackTraceElement getStackTraceElement(final ThreadInfo info) {
```

Comment:
Could you implement the following changes, please?
1. Rename this method to `getFirstStackTraceElement`
3. Return null on empty stack trace (instead of returning a dummy `StackTraceElement` with a magic value in it) and mark the method with `@Nullable` from JSpecify
4. Update above `case WAITING` and `case TIMED_WAITING` blocks accordingly, e.g.,

```
final StackTraceElement element = getFirstStackTraceElement(info);
final String className = element != null ? element.getClassName() : null;
final String method = element != null ? element.getMethodName() : null;
if ("java.lang.Object".equals(className) && "wait".equals(method)) {
    ...
```